src/Avalonia.Controls/TrayIcon.cs

```diff
@@ -123,9 +127,9 @@ static TrayIcon()
         public static readonly StyledProperty<bool> IsVisibleProperty =
             Visual.IsVisibleProperty.AddOwner<TrayIcon>();
 
-        public static void SetIcons(AvaloniaObject o, TrayIcons trayIcons) => o.SetValue(IconsProperty, trayIcons);
+        public static void SetIcons(Application o, TrayIcons trayIcons) => o.SetValue(IconsProperty, trayIcons);
```

Comment:
It should be strict to application only to avoid confusion, see https://github.com/AvaloniaUI/Avalonia/issues/7588